forward43/kickstarter_scraper.py

```diff
@@ -0,0 +1,104 @@
+import os
+
+import json
+import random
+import time
+import requests
+from pathlib import Path
+
+from datetime       import datetime
+from urllib.request import urlopen
+
+
+MAX_VALUE                  = 65536
+KICKSTARTER_URL            = 'https://www.kickstarter.com/discover/advanced.json?'
+KICKSTARTER_DEFAULT_PARAMS = ['sort=popularity']
+ATTRIBUTES                 = ['rank', 'id', 'name', 'creator', 'goal', 'pledged',
+                              'state', 'backers', 'launch_date', 'deadline', 'url']
+
+def write_data_to_file(data, category):
+    '''
+    Write json data to file
+    @data: dictionary
+    @category: category id
+    '''
+    filepath = '../data/{}.json'.format(category)
+
+    with open(filepath, 'w+') as f:
+        json.dump(data, f)
+
+
+def generate_url(category, page):
+    '''
+    Generates a URL to scrape from. It's necessary that category and page parameters are not None
+    '''
+    random_seed   = 'seed='   + str(random.randint(1, MAX_VALUE))
+    random_woe_id = 'woe_id=' + str(random.randint(1, MAX_VALUE))
```

Comment:
Interesting! Never heard of woe_id's before. 
A quick google showed that there's a lot more than that MAX_VALUE though: new york is `2459115` which is above the `MAX_VALUE`. And entering pseudorandom woe_id's myself gives me 0 results quite often, which might not be great given that we're dealing with rate-limiters. 

Maybe a way forward is to have a list of countries woe_id's and picking randomly from there, or setting the woe_id to 1 (everywhere, I believe?) and letting the random_seed do all the work of getting new projects

Reply:
For now, I'll set woe_id to 1. 

Getting a random woe_id should be possible for later.